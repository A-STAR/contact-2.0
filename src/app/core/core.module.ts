import { ErrorHandler, NgModule, Optional, SkipSelf } from '@angular/core';
import { DatePipe } from '@angular/common';

import { JwtHelper } from 'angular2-jwt';
import { TranslateService } from '@ngx-translate/core';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';

import { environment } from '../../environments/environment';

import { AuthEffects } from './auth/auth.effects';
import { AuthHttpService } from './auth/auth-http.service';
import { AuthService } from './auth/auth.service';
import { ContentTabService } from '../shared/components/content-tabstrip/tab/content-tab.service';
import { DataService } from './data/data.service';
import { DictionariesEffects } from './dictionaries/dictionaries.effects';
import { DictionariesService } from './dictionaries/dictionaries.service';
import { EntityTranslationsService } from './entity/translations/entity-translations.service';
import { ErrorHandlerService } from './error/error-handler.service';
import { LookupEffects } from './lookup/lookup.effects';
import { LookupService } from './lookup/lookup.service';
import { MenuService } from './menu/menu.service';
import { MetadataService } from './metadata/metadata.service';
import { NotificationsEffects } from './notifications/notifications.effects';
import { NotificationsService } from './notifications/notifications.service';
import { SettingsService } from './settings/settings.service';
import { ThemesService } from './themes/themes.service';
import { throwIfAlreadyLoaded } from './module-import-guard';
import { MetadataEffects } from './metadata/metadata.effects';
import { UserConstantsEffects } from './user/constants/user-constants.effects';
import { UserConstantsService } from './user/constants/user-constants.service';
import { UserDictionariesEffects } from './user/dictionaries/user-dictionaries.effects';
import { UserDictionariesService } from './user/dictionaries/user-dictionaries.service';
import { UserLanguagesEffects } from './user/languages/user-languages.effects';
import { UserLanguagesService } from './user/languages/user-languages.service';
import { UserPermissionsEffects } from './user/permissions/user-permissions.effects';
import { UserPermissionsService } from './user/permissions/user-permissions.service';
import { ValueConverterService } from './converter/value/value-converter.service';

import { rootReducer } from './state/root.reducer';

@NgModule({
  imports: [
    StoreModule.provideStore(rootReducer),
    EffectsModule.run(AuthEffects),
    EffectsModule.run(DictionariesEffects),
    EffectsModule.run(LookupEffects),
    EffectsModule.run(NotificationsEffects),
    EffectsModule.run(UserConstantsEffects),
    EffectsModule.run(UserDictionariesEffects),
    EffectsModule.run(UserLanguagesEffects),
    EffectsModule.run(UserPermissionsEffects),
    EffectsModule.run(MetadataEffects),
    environment.production
      ? []
      : StoreDevtoolsModule.instrumentOnlyWithExtension({
          maxAge: 1024
        })
  ],
  providers: [
    AuthHttpService,
    AuthService,
    ContentTabService,
    EntityTranslationsService,
    DictionariesService,
    DatePipe,
    DataService,
    JwtHelper,
    LookupService,
    MenuService,
    MetadataService,
    NotificationsService,
    SettingsService,
    ThemesService,
    TranslateService,
    UserConstantsService,
    UserDictionariesService,
    UserLanguagesService,
    UserPermissionsService,
    ValueConverterService,
    {
      provide: ErrorHandler,
      useClass: ErrorHandlerService
    }
  ],
})
export class CoreModule {
  constructor(@Optional() @SkipSelf() parentModule: CoreModule) {
    throwIfAlreadyLoaded(parentModule, 'CoreModule');
  }
}
