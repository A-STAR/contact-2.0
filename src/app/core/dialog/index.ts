export abstract class DialogFunctions {
  /**
   * NOTE: this prop cannot come from the prototype and has to be implemented in the children
   *
   * @abstract
   * @type {string}
   * @memberof DialogFunctions
   */
  abstract dialog: string;

  isDialog(dialog: string): boolean {
    return this.dialog === dialog;
  }

  setDialog(dialog: string = null): void {
    this.dialog = dialog;
  }

  closeDialog(): void {
    this.setDialog();
  }
  /**
   * Same as `closeDialog` just to maintain the `on` prefix used in EventEmitters
   *
   * @memberof DialogFunctions
   */
  onCloseDialog(): void {
    this.setDialog();
  }

  onResult(result: any): void {

  }
}
