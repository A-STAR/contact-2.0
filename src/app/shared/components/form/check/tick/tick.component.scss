.tick-wrapper {
  text-align: left;
}
.tick-icon {
  vertical-align: middle;
}
.tick-label {
  margin-left: 5px;
  cursor: pointer;
}
