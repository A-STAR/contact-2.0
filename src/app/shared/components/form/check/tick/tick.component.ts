import {
  ChangeDetectionStrategy,
  ChangeDetectorRef,
  Component,
  forwardRef,
  Input,
} from '@angular/core';
import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';

@Component({
  changeDetection: ChangeDetectionStrategy.OnPush,
  providers: [
    {
      provide: NG_VALUE_ACCESSOR,
      useExisting: forwardRef(() => TickComponent),
      multi: true,
    }
  ],
  selector: 'app-tick',
  styleUrls: ['./tick.component.scss'],
  template: `
  <div class="tick-wrapper">
    <i *ngIf="value" class="tick-icon icon co-checkbox-mark"></i>
    <span class="tick-label" (click)="onChange()"> {{ label | translate }}</span>
  </div>
  `
})
export class TickComponent implements ControlValueAccessor {
  @Input() label: string;

  private _disabled: boolean;
  value = false;

  constructor(
    private cdRef: ChangeDetectorRef,
  ) {}

  get disabled(): boolean {
    return this._disabled;
  }

  writeValue(value: boolean): void {
    this.value = value;
    this.cdRef.markForCheck();
  }

  registerOnChange(fn: Function): void {
    this.propagateChange = fn;
  }

  registerOnTouched(): void {
    // No need in touch callback for checkbox
    // because a click will change its value and mark control as dirty anyway
  }

  setDisabledState(disabled: boolean): void {
    this._disabled = disabled;
    this.cdRef.markForCheck();
  }

  onChange(): void {
    this.value = !this.value;
    this.propagateChange(this.value);
    this.cdRef.markForCheck();
  }

  private propagateChange: Function = () => {};
}
