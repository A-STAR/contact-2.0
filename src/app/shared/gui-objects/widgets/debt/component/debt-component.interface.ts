export interface IDebtComponent {
  id: number;
  typeCode: number;
  sum: number;
  currencyId: number;
}
