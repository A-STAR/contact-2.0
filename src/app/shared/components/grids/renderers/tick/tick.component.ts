import { ChangeDetectionStrategy, Component } from '@angular/core';
import { ICellRendererParams } from 'ag-grid/main';
import { ICellRendererAngularComp } from 'ag-grid-angular';

@Component({
  changeDetection: ChangeDetectionStrategy.OnPush,
  selector: 'app-grid-tick-renderer',
  template: `
    <div class="text-center">
      <i *ngIf="value" style="vertical-align: middle;" class="icon co-checkbox-mark"></i>
    </div>
  `,
})
export class TickRendererComponent implements ICellRendererAngularComp {
  private params: ICellRendererParams;

  agInit(params: ICellRendererParams): void {
    this.params = params;
  }

  refresh(): boolean {
    return false;
  }

  get value(): boolean {
    return this.params.value;
  }
}
