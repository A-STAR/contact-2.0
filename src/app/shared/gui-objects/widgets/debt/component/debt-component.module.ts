import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { DebtComponentGridModule } from './grid/debt-component-grid.module';

import { DebtComponentService } from './debt-component.service';

@NgModule({
  imports: [
    DebtComponentGridModule,
    CommonModule,
  ],
  exports: [
    DebtComponentGridModule,
  ],
  providers: [
    DebtComponentService,
  ]
})
export class DebtComponentModule { }
