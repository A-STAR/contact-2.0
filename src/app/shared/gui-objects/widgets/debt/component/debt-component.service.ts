import { Injectable } from '@angular/core';
import { Observable } from 'rxjs/Observable';

import { IDebtComponent } from './debt-component.interface';

import { DataService } from '../../../../../core/data/data.service';
import { NotificationsService } from '../../../../../core/notifications/notifications.service';

@Injectable()
export class DebtComponentService {
  static MESSAGE_DEBT_COMPONENT_SAVED = 'MESSAGE_DEBT_COMPONENT_SAVED';

  constructor(
    private dataService: DataService,
    private notificationsService: NotificationsService,
  ) {}

  fetchAll(debtId: number): Observable<Array<IDebtComponent>> {
    return this.dataService
      .read('/debts/{debtId}/components', { debtId })
      .map(response => response.components)
      .catch(this.notificationsService.error('errors.default.read').entity('entities.debtComponents.gen.plural').dispatchCallback());
  }

  fetch(debtId: number, debtComponentId: number): Observable<IDebtComponent> {
    return this.dataService
      .read('/debts/{debtId}/components/{debtComponentId}', { debtId, debtComponentId })
      .map(response => response.components[0])
      .catch(this.notificationsService.error('errors.default.read').entity('entities.debtComponents.gen.singular').dispatchCallback());
  }

  create(debtId: number, debtComponent: IDebtComponent): Observable<void> {
    return this.dataService
      .create('/debts/{debtId}/components', { debtId }, debtComponent)
      .catch(this.notificationsService.error('errors.default.create').entity('entities.debtComponents.gen.singular').dispatchCallback());
  }

  update(debtId: number, debtComponentId: number, debtComponent: Partial<IDebtComponent>): Observable<void> {
    return this.dataService
      .update('/debts/{debtId}/components/{debtComponentId}', { debtId, debtComponentId }, debtComponent)
      .catch(this.notificationsService.error('errors.default.update').entity('entities.debtComponents.gen.singular').dispatchCallback());
  }

  delete(debtId: number, debtComponentId: number): Observable<void> {
    return this.dataService
      .delete('/debts/{debtId}/components/{debtComponentId}', { debtId, debtComponentId })
      .catch(this.notificationsService.error('errors.default.delete').entity('entities.debtComponents.gen.singular').dispatchCallback());
  }
}
