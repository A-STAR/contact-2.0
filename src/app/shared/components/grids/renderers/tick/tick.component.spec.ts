import { async as Async, ComponentFixture, TestBed } from '@angular/core/testing';

import { TickRendererComponent } from './tick.component';

describe('TickRendererComponent', () => {
  let fixture: ComponentFixture<TickRendererComponent>;

  beforeEach(Async(() => {
    TestBed
      .configureTestingModule({
        declarations: [
          TickRendererComponent,
        ],
      })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TickRendererComponent);
  });

  it('should render tick renderer with false value', () => {
    fixture.componentInstance.agInit({ value: false } as any);
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it('should render tick renderer with true value', () => {
    fixture.componentInstance.agInit({ value: true } as any);
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
