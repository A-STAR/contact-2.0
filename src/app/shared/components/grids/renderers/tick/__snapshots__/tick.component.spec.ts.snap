// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TickRendererComponent should render tick renderer with false value 1`] = `
<div
  id="root0"
  ng-version="5.2.4"
>
  
    
  
  
  
</div>
`;

exports[`TickRendererComponent should render tick renderer with true value 1`] = `
<div
  id="root1"
  ng-version="5.2.4"
>
  
    
  
  <div
    class="text-center"
  >
    
      
    <i
      class="icon co-checkbox-mark"
      style="vertical-align: middle;"
    />
    
    
  </div>
  
  
</div>
`;
